packages/pyright-scip/snapshots/output/enclosing_range_class/advanced.py

```diff
@@ -0,0 +1,50 @@
+# < definition scip-python python snapshot-util 0.1 advanced/__init__:
+
+# format-options: showRanges
+# ⌄ start enclosing_range scip-python python snapshot-util 0.1 advanced/class_decorator().
```

Comment:
nit: This formatting is a bit off. Note that the definition line above uses a special `<` to indicate that the range begins at the start of the line, we should use similar logic for `begin` and `end` here.

Reply:
Agreed, although said syntax makes quick overview a bit harder. I would be happy to implement a standard to snapshots across scip-* repositories.

Reply:
> Agreed, although said syntax makes quick overview a bit harder

It's important that we have:
1. Consistent syntax for different source range annotations
2. Be able to mark the start of the line/end of the line.

Another alternative is to just offset everything by +2, which is done in some places, but it makes browsing the snapshot files more annoying in an editor because of the red squiggles.

That said, I'm not married to the exact syntax being used (if you have other suggestions, I'm all ears). The important criteria are 1 and 2 above.

---

> I would be happy to implement a standard to snapshots across scip-* repositories.

While we do have some core functionality in `scip snapshot` here: https://github.com/sourcegraph/scip/blob/main/docs/CLI.md#scip-snapshot

We don't use it in the other indexer repositories because:
1. The various indexers are in different languages, so we can't reuse the snapshotting logic as a library easily (it is in Go).
4. It's valuable to quickly be able to customize the snapshotting logic depending on the use-case without having to change it in a central place.

It's not that much code overall, about 100-200 lines, but having it be different allows moving different repos in a more de-coupled way.